### ISSUE TITLE
Mapped types expand correctly but fail field access (`Undefined field`)

### ISSUE THREAD

**Issue by saffr3n** [EmmyLuaLs/emmylua-analyzer-rust#917]
EmmyLua correctly expands mapped types (`[K in keyof T]`) for type display.
However, the expanded fields are not usable for field access, resulting in false diagnostics.

This happens even without optional modifiers, so the issue is not related to `-?`.

## Minimal Reproduction

```lua
---@alias Required<T> { [K in keyof T]: T[K] -?; }

---@alias SomeMap { some_int?: integer, some_str?: string }

---@type Required<SomeMap>
local a -- `a` is correctly inferred as `Required<SomeMap> = { some_int: integer, some_str: string }`
local _ = a.some_int -- Diagnostic: Undefined field `some_int`.
```

## Expected Behavior

Field access like `a.some_int` should work.

## Actual Behavior

Hover / inferred type shows the expanded structure correctly, but field access fails.
